src/net/sourceforge/kolmafia/request/UseSkillRequest.java

```diff
@@ -893,20 +862,18 @@ public static final void optimizeEquipment(final int skillId) {
       }
     }
 
-    switch (skillId) {
-      case SkillPool.INVISIBLE_AVATAR, SkillPool.TRIPLE_SIZE -> equipForSkill(
-          skillId, UseSkillRequest.POWERFUL_GLOVE, UseSkillRequest.REPLICA_POWERFUL_GLOVE);
-      case SkillPool.MAKE_SWEATADE,
-          SkillPool.DRENCH_YOURSELF_IN_SWEAT,
-          SkillPool.SIP_SOME_SWEAT -> equipForSkill(
-          skillId,
-          UseSkillRequest.DESIGNER_SWEATPANTS,
-          UseSkillRequest.REPLICA_DESIGNER_SWEATPANTS);
-      case SkillPool.CINCHO_DISPENSE_SALT_AND_LIME,
-          SkillPool.CINCHO_PARTY_SOUNDTRACK,
-          SkillPool.CINCHO_FIESTA_EXIT -> equipForSkill(
-          skillId, UseSkillRequest.CINCHO_DE_MAYO, UseSkillRequest.REPLICA_CINCHO_DE_MAYO);
-      case SkillPool.REST_UPSIDE_DOWN -> equipForSkill(skillId, UseSkillRequest.BAT_WINGS);
+    if (SkillDatabase.getSkillTags(skillId).contains(SkillDatabase.SkillTag.NONCOMBAT)) {
```

Comment:
if you only have the replica powerful glove and try to `cast triple size`, what happens? Does mafia error once and then succeed?

if you only have the normal powerful glove does it succeed, then error?

Reply:
Hm ok I will split this stream and see if we have equipped or can equip anything that matches the filter.